indexer/lib/advertisement-walker.js

```diff
@@ -252,23 +260,43 @@ export async function fetchAdvertisedPayload (providerAddress, advertisementCid,
     return { previousAdvertisementCid }
   }
 
-  const entriesChunk =
+  const meta = parseMetadata(advertisement.Metadata['/'].bytes)
+  const pieceCid = meta.deal?.PieceCID.toString()
+
+  try {
+    const entriesChunk =
     /** @type {{
      Entries: { '/' :  { bytes: string } }[]
     }} */(
-      await fetchCid(providerAddress, entriesCid, { fetchTimeout })
-    )
-  debug('entriesChunk %s %j', entriesCid, entriesChunk.Entries.slice(0, 5))
-  const entryHash = entriesChunk.Entries[0]['/'].bytes
-  const payloadCid = CID.create(1, 0x55 /* raw */, multihash.decode(Buffer.from(entryHash, 'base64'))).toString()
+        await fetchCid(providerAddress, entriesCid, { fetchTimeout })
+      )
+    debug('entriesChunk %s %j', entriesCid, entriesChunk.Entries.slice(0, 5))
+    const entryHash = entriesChunk.Entries[0]['/'].bytes
+    const payloadCid = CID.create(1, 0x55 /* raw */, multihash.decode(Buffer.from(entryHash, 'base64'))).toString()
 
-  const meta = parseMetadata(advertisement.Metadata['/'].bytes)
-  const pieceCid = meta.deal?.PieceCID.toString()
-
-  return {
-    previousAdvertisementCid,
-    pieceCid,
-    payloadCid
+    return {
+      previousAdvertisementCid,
+      pieceCid,
+      payloadCid
+    }
+  } catch (err) {
+    if (err && typeof err === 'object' && 'statusCode' in err && err.statusCode === 404) {
+      // The index provider cannot find the advertised entries. We cannot do much about that,
+      // it's unlikely that further request will succeed. Let's skip this advertisement.
+      debug(
+        'Cannot fetch ad %s entries %s: %s %s',
+        advertisementCid,
+        entriesCid,
+        err.statusCode,
+        /** @type {any} */(err).serverMessage ?? '<not found>'
+      )
+      return {
+        entriesFetchError: true,
```

Comment:
This is not the best property name, can you suggest a better one? Or may I should set the value to a string instead of an error?

```suggestion
        entriesFetchError: String(err.statusCode),
```

Reply:
I was just looking up this property, as was assuming it's one of these
- an `Error`
- an array of entries for which there was a fetch error
- the count of entries for which there was a fetch error

What about `entriesFetchError: err`? Or otherwise `hasEntriesFetchError: true`